go-controller/pkg/cni/cni.go

```diff
@@ -148,12 +183,45 @@ func (pr *PodRequest) cmdAdd(kubeAuth *KubeAPIAuth, clientset *ClientSet) (*Resp
 
 	podInterfaceInfo.SkipIPConfig = kubevirt.IsPodLiveMigratable(pod)
 
+	var userDefinedNetPodInfo *PodInterfaceInfo
+	if hasUserDefinedPrimaryNetwork {
+		klog.Infof("About to plumb the user defined NET for %q on pod %q", userDefinedNetworkName, podName)
+		userDefinedNetPodInfo, err = PodAnnotation2PodInfo(
+			annotations,
+			userDefinedPrimaryNetworkAnnotation,
+			pr.PodUID,
+			"",
+			userDefinedNetworkName, // TODO: we need to find a way to have here access to the network name
+			userDefinedNetworkName, // TODO: we can have the NAD name from the annotations. But the network name is missing. maybe we can add that to the annotations
+			pr.CNIConf.MTU,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+	}
+
 	response := &Response{KubeAuth: kubeAuth}
 	if !config.UnprivilegedMode {
 		response.Result, err = pr.getCNIResult(clientset, podInterfaceInfo)
 		if err != nil {
 			return nil, err
 		}
+		if hasUserDefinedPrimaryNetwork && userDefinedNetPodInfo != nil {
+			klog.Infof("creating the user defined net request for %q on pod %q", userDefinedNetworkName, podName)
+			userDefinedRequest := newUserDefinedPrimaryNetworkPodRequest(
+				pr.Command,
+				pod,
+				pr.SandboxID,
+				pr.Netns,
+				"sdn1", // TODO: need to find a name here. This one is OK I guess
+				userDefinedNetworkName,
+				userDefinedNetPodInfo.NADName,
+			)
+			if _, err := userDefinedRequest.getCNIResult(clientset, userDefinedNetPodInfo); err != nil {
+				return nil, err
+			}
+		}
```

Comment:
if we're on unpriviledged more, this code right here **must** happen in the CNI shim.

This means that probably the CNI server side (running in ovnkube-controller) should reply to the client (shim) w/ whatever extra info it needs, thus avoiding reading more stuff on the shim.